tensorflow/compiler/xla/service/gpu/gemm_rewriter.cc

```diff
@@ -101,39 +101,100 @@ bool IsF8Type(const HloInstruction *instr) {
   }
 }
 
-bool IsF8TypeRecursiveImpl(const HloInstruction *instr,
-                           absl::flat_hash_set<int> &visited_instrs) {
+// Recursively collects unary, pad, divide or multiply operands of instr until
+// an instruction with FP8 element type is reached. Returns std::nullopt when no
+// FP8 instruction is reached.
+std::optional<std::vector<HloInstruction *>> FindF8SubgraphRecursive(
+    const HloInstruction *instr, absl::flat_hash_set<int> &visited_instrs,
+    std::vector<HloInstruction *> subgraph) {
   // Avoid visiting the same instruction more than once.
   if (!visited_instrs.emplace(instr->unique_id()).second) {
-    return false;
+    return std::nullopt;
   }
+  subgraph.emplace_back(const_cast<HloInstruction *>(instr));
   if (IsF8Type(instr)) {
-    return true;
+    return subgraph;
   } else {
     if (instr->operand_count() == 1 || instr->opcode() == HloOpcode::kDivide ||
         instr->opcode() == HloOpcode::kPad) {
-      return IsF8TypeRecursiveImpl(instr->operand(0), visited_instrs);
+      return FindF8SubgraphRecursive(instr->operand(0), visited_instrs,
+                                     subgraph);
     } else if (instr->opcode() == HloOpcode::kMultiply) {
-      return IsF8TypeRecursiveImpl(instr->operand(0), visited_instrs) ||
-             IsF8TypeRecursiveImpl(instr->operand(1), visited_instrs);
-    } else {
-      return false;
+      for (int k = 0; k < 2; ++k) {
+        auto mult_subgraph = FindF8SubgraphRecursive(instr->operand(k),
+                                                     visited_instrs, subgraph);
+        if (mult_subgraph.has_value()) {
+          return mult_subgraph;
+        }
+      }
     }
+    return std::nullopt;
   }
 }
 
-bool IsF8TypeRecursive(const HloInstruction *instr) {
+// Returns whether instr and its operands describe a pattern that can be
+// rewritten into an FP8 Custom Call. If applicable, captures the operand of the
```

Comment:
The first sentence isn't entirely accurate, as `inst` only describes part of the pattern that can be rewritten into an FP8 Custom Call. Replace the first sentence with something like:

> Returns whether instr, which should be an input to a Dot, matches a pattern that allows the Dot to be rewritten into an F8 custom call.